www/class/centreonCustomView.class.php

```diff
@@ -481,30 +529,42 @@ public function deleteCustomView($viewId)
     /**
      * Update Custom View
      *
-     * @param $params
-     * @return mixed
+     * @param int $customViewId
+     * @param string $name
+     * @param string $layout
+     * @param int $public
+     * @param bool $permission
+     * @param bool $authorized
+     * @return int $customViewId
      * @throws Exception
      */
-    public function updateCustomView($params)
-    {
-        if ($this->checkPermission($params['custom_view_id']) === true) {
-            $public = 0;
-            if (isset($params['public'])) {
-                $public = $params['public'];
-            }
-            $query = 'UPDATE custom_views SET `name` = :viewName, `layout` = :layout, `public` = :typeView ' .
-                'WHERE `custom_view_id` = :viewId';
-            $stmt = $this->db->prepare($query);
-            $stmt->bindParam(':viewName', $params['name'], PDO::PARAM_STR);
-            $stmt->bindParam(':layout', $params['layout']['layout'], PDO::PARAM_STR);
-            $stmt->bindParam(':typeView', $public, PDO::PARAM_INT);
-            $stmt->bindParam(':viewId', $params['custom_view_id'], PDO::PARAM_INT);
-            $dbResult = $stmt->execute();
-            if (!$dbResult) {
-                throw new \Exception("An error occured");
-            }
+    public function updateCustomView(
+        int $customViewId,
+        string $name,
+        string $layout,
+        int $public,
```

Comment:
```suggestion
        ?int $public,
```
Typehint should be ?int as the expected value could be 0/1/null